p2p/src/collection/mod.rs

```diff
@@ -0,0 +1,70 @@
+//! An [Originator] sends messages out to a set of [Endpoint]s and collects responses.
+//!
+//! The originator's [Collector] is used to distribute and collect responses.
+//! The endpoint's [Collector] is used to receive and respond to messages.
+
+use bytes::Bytes;
+use commonware_cryptography::PublicKey;
+use futures::channel::oneshot;
+use std::{collections::HashMap, fmt::Debug, future::Future};
+
+pub mod actor;
+
+/// Trait for an item that has an associated, unique ID.
+pub trait Idable: Clone + Send + 'static {
+    type ID: Debug + Send + 'static;
+
+    /// Returns the unique ID of the item.
+    fn id(&self) -> Self::ID;
+}
+
+/// Interface for actor that will disperse messages and collect responses.
+pub trait Collector: Clone + Send + 'static {
+    type Message: Idable + Debug + Send + 'static;
```

Comment:
The traits for this should be `Digestible` and `Commitable`: https://github.com/commonwarexyz/monorepo/blob/da9ba1498f7d2906d88b8a3d4f733e267eb4ac2f/broadcast/src/buffered/ingress.rs#L11